byway/src/main/webapp/home.js

```diff
@@ -86,22 +87,44 @@ function showIncompleteTrip(tripNum, trip, isDestinationsMissing, isInterestsMis
  * @param {Number} tripNum 
  * @param {Trip} trip
  */
-function showCompleteTrip(tripNum, trip){
+async function showCompleteTrip(trip){
   let container = document.getElementById("past-trips-container");
   let pastTrip = document.createElement('div');
   pastTrip.className = "past-trip";
   let title = document.createElement('a');
+  title.id = "title-" + trip.keyString;
   pastTrip.append(title);
   let mapContainer = document.createElement('div');
   mapContainer.className = 'map';
-  mapContainer.id = "map-" + trip.keyString
+  mapContainer.id = "map-" + trip.keyString;
   pastTrip.append(mapContainer);
   container.append(pastTrip);
   initMap(trip.start, trip.start, trip.route, trip.keyString);
-  title.innerText = "Trip #" + tripNum;
+  await constructTripTitle(trip, trip.keyString);
   title.href = configureTripKeyForPath(trip.keyString, "/routepage.html");
 }
 
+async function constructTripTitle(trip, keyString){
+  let title = document.getElementById('title-' + keyString);
+  title.innerText = "";
+  for(let i = 0; i<10; i++){
```

Comment:
Is there a reason you iterate `10` times? It seems like you constructTripTitle per complete trip, so not sure why you need to loop. If I understood wrong (i.e. it actually loops through all trips in this stingle function), then you might want to change the `10` to the size of the trips saved for the user

Reply:
so basically, this is just looping until the findPlace doesn't return an error. I wasn't sure which number to set it at bc sometimes it doesn't need to be delated and sometimes it needs to be delayed more than 1 second